src/main/resources/org/jenkinsci/test/acceptance/docker/fixtures/GitContainer/Dockerfile

```diff
@@ -23,8 +23,14 @@ RUN useradd git -d /home/git && \
 ADD unsafe.pub /home/git/
 RUN cat /home/git/unsafe.pub >> /home/git/.ssh/authorized_keys
 
-# run SSHD in the foreground with error messages to stderr
-CMD /usr/sbin/sshd -D -e
-
 # give the whole /home/git back to the git user
 RUN chown -R git /home/git
+# Proper permissions for ssh
+RUN chown -R git:git /home/git/.ssh
+RUN chmod 700 /home/git/.ssh
+RUN chmod 600 /home/git/.ssh/authorized_keys
+
+RUN echo "HostbasedAcceptedKeyTypes +ssh-rsa\nHostKeyAlgorithms +ssh-rsa\nPubkeyAcceptedKeyTypes +ssh-rsa" >> /etc/ssh/sshd_config
```

Comment:
We need this for jsch to be able to sftp into the container, I tried with more modern algorithms for the keys but jsch did not supported them